### PR TITLE
Convert nRF9160 samples to use the new k_work API

### PR DESCRIPTION
**Pull request by anangl**
Convert the following nRF9160 samples to use the new k_work API:
- A-GPS
- UDP
- LTE Sensor Gateway
- Cloud client
- AWS IoT
- Azure FOTA
- Azure IoT Hub
- LwM2M Client

Ref: NCSDK-9417

### CODE REVIEW

File: samples/nrf9160/lwm2m_client/src/lwm2m/lwm2m_accelerometer.c [hunk 1]
```diff
@@ -259,7 +259,7 @@ int lwm2m_init_accel(void)
 				  &timestamp, sizeof(timestamp), 0);
 
 	if (IS_ENABLED(CONFIG_FLIP_POLL)) {
-		k_delayed_work_submit(&flip_poll_work, K_NO_WAIT);
+		k_work_schedule(&flip_poll_work, K_NO_WAIT);
```
Comment (pabigot) [inline samples/nrf9160/lwm2m_client/src/lwm2m/lwm2m_accelerometer.c:262]: There are several like this where it looks like `k_work_reschedule()` might be appropriate, to cut short the delay if the work is already scheduled.  But maybe that can't happen.

File: samples/nrf9160/lwm2m_client/src/ui/led_pwm.c [hunk 2]
```diff
@@ -152,7 +153,7 @@ static void led_update(struct led *led)
 		int32_t next_delay =
 			led->effect->steps[led->effect_step].substep_time;
 
-		k_delayed_work_submit(&led->work, K_MSEC(next_delay));
+		k_work_schedule(&led->work, K_MSEC(next_delay));
```
Comment (pabigot) [inline samples/nrf9160/lwm2m_client/src/ui/led_pwm.c:156]: This code seems like it's similar to the version discussed at https://github.com/nrfconnect/sdk-nrf/pull/4454#discussion_r630955328, though it doesn't have the potential null pointer dereference.  Here I notice that use of `schedule` means it delays the change in effect until the work is next executed, if a previous effect was active.

Maybe this LED functionality should be pulled out into a library that the samples (and applications) can use.  (Or maybe not; I'm only looking at the diffs here.)

**Review by pabigot — APPROVED** [review]
Seems fine, my concerns that `reschedule` may be more appropriate for some places where `schedule` is now used aren't worth delaying approval.

File: samples/nrf9160/lwm2m_client/src/lwm2m/lwm2m_accelerometer.c (hunk 1 above)
Comment (anangl) [reply samples/nrf9160/lwm2m_client/src/lwm2m/lwm2m_accelerometer.c:262]: Here the module is initialized and the work is scheduled for the first time. Then it is scheduled again only from its own handler. That's why I used `k_work_schedule()` in both these cases.

**Review by rlubos — APPROVED** [review]
LGTM

**Comment by anangl** [comment]
Rebased.

**Review by jtguggedal — APPROVED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Pull request merged**